### PR TITLE
Add Snaplet data generation to docs.

### PR DESCRIPTION
**Pull request by peterp**
I've added Snaplet's new data generation mode to the "Seed your database" during local development block.

### CODE REVIEW

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 1]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+              count: 3,
+            }
+          }
+        })
+      }
+   }
+})
+```
+
+Running `npx snaplet generate` would produce these SQL statements:
```
Comment (justinvdm) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:113]: _Bikeshed_: Not sure if it matters enough to change, but its more that we're running these statements against the database than simply producing these statements (we're actually modify the database).

**Review by justinvdm — APPROVED** [review]
Looks great :) really well explained

File: apps/docs/pages/guides/cli/seeding-your-database.mdx (hunk 1 above)
Comment (peterp) [reply apps/docs/pages/guides/cli/seeding-your-database.mdx:113]: We'll add a `--sql` flag to the command, it's just not in there at the moment.

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 2]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+
+INSERT INTO "User" (name, email) VALUES ("Steven Plank", "s@plank.org")
+INSERT INTO "Comment" (text, userId, postId) VALUES ("Actually, that's not correct...", 4, 1)
+```
+
+A benefit of using Snaplet is that _so much_ of the resulting data has been contextually generated. A minimum definition of intention still results in a comprehensive set of data returned.
 
-## Generating seed data from Production
+Three things that were automatically generated:
+- Three comments were requested, resulting in the `text` and three additional users.
```
Comment (peterp) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:139]:
```suggestion
- Three comments were requested, resulting in the generated comment `text`, and three additional users.
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx (hunk 1 above)
Comment (peterp) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:113]:
```suggestion
Running `npx snaplet generate` would modify your database with these SQL statements:
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 3]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+              count: 3,
+            }
+          }
+        })
+      }
+   }
+})
+```
+
+Running `npx snaplet generate` would modify your database with these SQL statements:
```
Comment (peterp) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:113]:
```suggestion
Running `npx snaplet generate` would modify your target database with these SQL statements:
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 4]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+              count: 3,
+            }
+          }
+        })
+      }
+   }
+})
+```
+
+Running `npx snaplet generate` would modify your target database with these SQL statements:
```
Comment (peterp) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:113]:
```suggestion
Running `npx snaplet generate --sql > seed.sql` would generate these SQL statements:
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 5]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
 ```sql supabase/seed.sql
 COPY countries (name, code) FROM 'seed/countries.csv' WITH (FORMAT csv);
 ```
+\*/}
+
+## Seed your database with Snaplet
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:53]:
```suggestion
## Generating seed data
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 6]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+
+Snaplet generates seed data from a database schema and a definition of the intended values. For instance if you wanted to seed a typical blog application, with the following schema:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:55]:
```suggestion
You can generate seed data for local development using [Snaplet](https://www.snaplet.dev/).

Let's generate some seed data for a typical blog application, with the following schema:
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 7]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+  content text,
+  "createdBy" bigint references "User" (id)
+);
+```
+\*/}
+
+And the intended values you want to generate:
+  * A `Post` with the title `"There is a lot of snow around here!"`
+  * The `Post.createdBy` user with an email address ending in `"@acme.org"`
+  * Three `Post.comments` from three different users.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:88]:
```suggestion
We can use the Snaplet configuration file to define the values you want to generate. For example

* A `Post` with the title `"There is a lot of snow around here!"`
* The `Post.createdBy` user with an email address ending in `"@acme.org"`
* Three `Post.comments` from three different users.
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 8]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+
+The definition of intended values would resemble:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:90]:
```suggestion

```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 9]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+              count: 3,
+            }
+          }
+        })
+      }
+   }
+})
+```
+
+Running `npx snaplet generate --sql > seed.sql` would generate these SQL statements:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:113]:
```suggestion
Running `npx snaplet generate --sql > supabase/seed.sql` would generate these relevant SQL statements inside our `supabase/seed.sql` file:
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 10]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+INSERT INTO "Comment" (text, userId, postId) VALUES ("I love cheese", 2, 1)
+
+INSERT INTO "User" (name, email) VALUES ("John Rambo", "rambo@trymore.dev")
+INSERT INTO "Comment" (text, userId, postId) VALUES ("Lorem ipsum dolar sit", 3, 1)
+
+INSERT INTO "User" (name, email) VALUES ("Steven Plank", "s@plank.org")
+INSERT INTO "Comment" (text, userId, postId) VALUES ("Actually, that's not correct...", 4, 1)
+```
+
+A benefit of using Snaplet is that _so much_ of the resulting data has been contextually generated. A minimum definition of intention still results in a comprehensive set of data returned.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:136]:
```suggestion

```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 11]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
 
-## Generating seed data from Production
+Three things that were automatically generated:
+- Three comments were requested, resulting in the generated comment `text`, and three additional users.
+- The `content` of `Post` was generated
+- The `Post.createBy` user was requested, the `name` was generated.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:141]:
```suggestion
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 12]
```diff
@@ -49,11 +48,99 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
 
-- Use snaplet
-  \*/}
+For more information about how this works reference [Snaplet's documentation](https://docs.snaplet.dev/references/data-operations/generate)
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:143]:
```suggestion
For more information, check out Snaplet's ["generate" documentation](https://docs.snaplet.dev/references/data-operations/generate)
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 13]
```diff
@@ -49,11 +48,94 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+
+
+create table "Post" (
+  id bigint not null primary key,
+  title text not null,
+  content text,
+  "createdBy" bigint references "User" (id)
+);
+```
+\*/}
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:85]:
```suggestion
```

File: apps/docs/pages/guides/cli/seeding-your-database.mdx [hunk 14]
```diff
@@ -49,11 +48,68 @@ You can use the `COPY` command to import CSV data into your database. For exampl
...
+Let's generate some seed data for a typical blog application, with the following schema:
+
+![An example schema](/docs/img/guides/cli/snaplet-example-schema.png)
+
+We can use the Snaplet configuration file to define the values you want to generate. For example
+
+* A `Post` with the title `"There is a lot of snow around here!"`
+* The `Post.createdBy` user with an email address ending in `"@acme.org"`
+* Three `Post.comments` from three different users.
+```ts snaplet.config.ts
```
Comment (kiwicopple) [inline apps/docs/pages/guides/cli/seeding-your-database.mdx:66]:
```suggestion
* Three `Post.comments` from three different users.

```ts snaplet.config.ts
```

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**